osu.Android/Properties/AndroidManifest.xml

```diff
@@ -1,10 +1,176 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" package="sh.ppy.osulazer" android:installLocation="auto" android:versionName="0.1.0">
+<manifest 
+  xmlns:android="http://schemas.android.com/apk/res/android" 
+  android:versionCode="1" 
+  package="sh.ppy.osulazer" 
+  android:installLocation="auto" 
+  android:versionName="0.1.0">
 	<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="28" />
 	<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
 	<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
 	<uses-permission android:name="android.permission.WAKE_LOCK" />
 	<uses-permission android:name="android.permission.READ_FRAME_BUFFER" />
 	<uses-permission android:name="android.permission.INTERNET" />
-	<application android:allowBackup="true" android:supportsRtl="true" android:label="osu!" android:icon="@drawable/lazer" />
+  <application
+    android:allowBackup="true"
+    android:supportsRtl="true"
+    android:label="osu!"
+    android:icon="@drawable/lazer" >
+
+    <activity
+      android:theme="@android:style/Theme.NoTitleBar"
+      android:name="md59c875562ee5070384021e1e981f7bd4e.OsuGameActivity"
+      android:screenOrientation="fullSensor"
+      android:configChanges="orientation|screenSize"
+      android:hardwareAccelerated="true"
+      android:supportsPictureInPicture="false">
+
+      <intent-filter>
+        <action
+            android:name="android.intent.action.MAIN" />
+        <category
+            android:name="android.intent.category.LAUNCHER" />
+      </intent-filter>
+      
+      <intent-filter>
+        <action
+            android:name="android.intent.action.VIEW" />
+        <category
+            android:name="android.intent.category.DEFAULT" />
+        <category
+            android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="file" />
+        <data android:mimeType="*/*" />
+        <data
+            android:pathPattern=".*\.osz" />
```

Comment:
this seems a bit hacky. is there no better way of doing this?

Reply:
i removed some useless lines, now it support firefox and some file explorers.
but in android, different apps may use different way to share files, i am not sure if other apps are supported

Reply:
i have tested it with qq, firefox, es file explorer and chrome.
qq, firefox and es works properly, but chrome doesn't expose the real file path.

Reply:
Much better, thanks. Would be interesting to see how other apps work with chrome (maybe search github for other android apps which do file importing using a keyword from the metadata file?)

Reply:
the official way to read the files shared by chrome is use ContentResolver to open a input stream.
yes, we can open a input stream from it, but can't get the file path of it

Reply:
we can force covert it to real file path, but if the user use a custom download path in chrome the converting will be wrong

Reply:
An import stream should be fine for our purposes.

Reply:
is there any existing api to import osz from a tream?

Reply:
Can be added very easily.

Reply:
If we had stream support, would it replace what you have in this PR as a better option? Or would it be in addition? If you show me where the stream is received and what format it is in, I can setup the handling at osu!'s end for you.

Reply:
now i pass all filepaths to OsuGameAndroid then import them in LoadComplete.
the stream i open is a System.IO.Stream, and it's the same as a stream opened with a osz file.

Reply:
it will only be in addition because only chrome uses this way to share files (or other app share file using "content://.......")
and now i auto scan the download directory when open osu,
 so it's not necessary to let the user open osz in chrome :)

Reply:
it is necessary still beacause you can chose beetwen downloading and opening in app @EdrowsLuo

Reply:
I did some research a while ago (https://github.com/Dragicafit/osu/commit/2f829ccc072713daf68d67ce38f31e5077701320) and I had to implement this disgusting line to support my file explorer:
`DataPathPatterns = new string[] { @".*\\.osz", @".*\\..*\\.osz", @".*\\..*\\..*\\.osz", @".*\\..*\\..*\\..*\\.osz", @".*\\..*\\..*\\..*\\..*\\.osz", @".*\\..*\\..*\\..*\\..*\\..*\\.osz", }`
(I don't know if your way works)
I found the solution at:
https://stackoverflow.com/questions/3760276/android-intent-filter-associate-app-with-file-extension/12153391#12153391

Reply:
@Dragicafit can you share me your file explorer?

Reply:
the file manager with lineageOS 15.1

Reply:
It's explained herehttps://stackoverflow.com/a/39885806

Reply:
@Dragicafit ok, i understand

Reply:
And implement https://github.com/ppy/osu/issues/1705 should be better